### PR TITLE
gh#23 - Fixing compilation warning in ut-core. 

### PR DESCRIPTION
**Pull request by kanjoe24**
Fix for compilation warning in ut-core. 
Also created patch to fix compilation warnings in CUnit folder.

### CODE REVIEW

**Comment by mhughesacn** [comment]
Hi Kanjoe: There is an actual code match here to the cUnit framework.  TestRun.patch contains TestRun.c which comes from cUnit.  cUnit is LGPL licensed and we need to be careful about keeping Apache and LGPL material separate.  As this patch contains only 1 file, please can the entire patch be moved under the src/cunit/cunit_lgpl directory?

Also, the patch needs a header with From/Date/ fields, and especially a Source: field which in this case should credit (some of) the code to cunit.

**Comment by kanjoe24** [comment]
> Hi Kanjoe: There is an actual code match here to the cUnit framework. TestRun.patch contains TestRun.c which comes from cUnit. cUnit is LGPL licensed and we need to be careful about keeping Apache and LGPL material separate. As this patch contains only 1 file, please can the entire patch be moved under the src/cunit/cunit_lgpl directory?

Sure, will do this.
> Also, the patch needs a header with From/Date/ fields, and especially a Source: field which in this case should credit (some of) the code to cunit.

Can you give me an example please, which I can look at and follow.

**Comment by mhughesacn** [comment]
Hi Kanjoe:
Look e.g. in meta-rdk-broadband on gerrit to see examples (usually patches are in the meta layers).  Here is one:
https://code.rdkcentral.com/r/plugins/gitiles/rdk/components/generic/rdk-oe/meta-rdk-broadband/+/refs/heads/rdk-next/recipes-rdkb/quagga/files/quagga_crash_fix.patch
but there are lots - all I really want is that the patch give some credit if it isn't your own code.  Source: Comcast, Sky etc.or Source: backport from opensource etc.

File: src/cunit/patches/TestRun.patch [hunk 1]
```diff
@@ -0,0 +1,69 @@
+--- CUnit-2.1-3/CUnit/Sources/Framework/TestRun.c	2024-04-05 17:03:33.892651597 +0100
```
Comment (Ulrond) [inline src/cunit/patches/TestRun.patch:1]: ah this patch is the wrong way round, all the - should be + and all the + should be minus..

So your left & right are in the wrong place I think.

**Review by Ulrond — CHANGES_REQUESTED** [review]
think your patch is incorrect.

**Comment by mhughesacn** [comment]
Sorry Kanjoe, I should have just specified the fields I wanted in the patch header:

From: Your email
Subject: You decide!
Date: Today's date or something sensible
Source: Uses code from the CUnit Testing Framework

File: build.sh [hunk 2]
```diff
@@ -31,5 +31,8 @@ else
     wget https://sourceforge.net/projects/cunit/files/CUnit/2.1-3/CUnit-2.1-3.tar.bz2 --no-check-certificate -P framework/
     tar xvfj framework/CUnit-2.1-3.tar.bz2 -C ./framework/
     cp framework/CUnit-2.1-3/CUnit/Headers/CUnit.h.in framework/CUnit-2.1-3/CUnit/Headers/CUnit.h
+    cp src/cunit/patches/TestRun.patch framework/.
```
[the remark this answers is not in the document]
Comment (kanjoe24) [reply build.sh]: Done.

File: src/cunit/patches/TestRun.patch (hunk 1 above)
Comment (kanjoe24) [reply src/cunit/patches/TestRun.patch]: Done.

File: build.sh [hunk 3]
```diff
@@ -31,5 +31,8 @@ else
...
+    cd framework/
+    patch -u -R -b CUnit-2.1-3/CUnit/Sources/Framework/TestRun.c -i TestRun.patch
```
[the remark this answers is not in the document]
Comment (kanjoe24) [reply build.sh]: Done

**Comment by kanjoe24** [comment]
> Sorry Kanjoe, I should have just specified the fields I wanted in the patch header:
> 
> From: Your email Subject: You decide! Date: Today's date or something sensible Source: Uses code from the CUnit Testing Framework

Thanks, I have added it now. Can you have a look please.

File: src/cunit/cunit_lgpl/patches/CorrectBuildWarningsInCunit.patch [hunk 4]
```diff
@@ -0,0 +1,82 @@
+#Date: April 08, 2024
+#Subject: [PATCH] Patch to fix compilation warning
+#		 in TestRun.c under CUnit directory. 
+#
+#Source: Uses code from the CUnit Testing Framework 
+#Notice: This software is licensed by RDK Management under the GNU General
```
[the remark this answers is not in the document]
Comment (kanjoe24) [reply src/cunit/cunit_lgpl/patches/CorrectBuildWarningsInCunit.patch]: Done.

**Review by Ulrond — APPROVED** [review]

**Pull request merged**